src/test/java/seedu/address/commons/util/StringUtilTest.java

```diff
@@ -45,6 +45,86 @@ public void isNonZeroUnsignedInteger() {
         assertTrue(StringUtil.isNonZeroUnsignedInteger("10"));
     }
 
+    //---------------- Tests for isIntegerUpperBounded --------------------------------------
+
+    @Test
+    public void isIntegerUpperBounded() {
+        final int validLimit = 1000;
+
+        // EP: empty strings
+        assertFalse(StringUtil.isIntegerUpperBounded("", validLimit)); // Boundary value
+        assertFalse(StringUtil.isIntegerUpperBounded("  ", validLimit));
+
+        // EP: not a number
+        assertFalse(StringUtil.isIntegerUpperBounded("a", validLimit));
+        assertFalse(StringUtil.isIntegerUpperBounded("aaa", validLimit));
+
+        // EP: zero
+        assertTrue(StringUtil.isIntegerUpperBounded("0", validLimit));
+
+        // EP: negative number
+        assertTrue(StringUtil.isIntegerUpperBounded("-1", validLimit));
+
+        // EP: zero as prefix
+        assertTrue(StringUtil.isIntegerUpperBounded("01", validLimit));
+
+        // EP: positive number smaller than limit
+        assertTrue(StringUtil.isIntegerUpperBounded(Integer.toString(validLimit - 1), validLimit));
+
+        // EP: number equal to limit
+        assertTrue(StringUtil.isIntegerUpperBounded(Integer.toString(validLimit), validLimit));
+
+        // EP: number above limit
+        assertFalse(StringUtil.isIntegerUpperBounded(Integer.toString(validLimit + 1), validLimit));
+
+        // EP: numbers with white space
+        assertFalse(StringUtil.isIntegerUpperBounded(" 10 ", validLimit)); // Leading/trailing spaces
```

Comment:
which kind of exception will occur if there are leading or trailing spaces for index

Reply:
```NumberParseException```, same with all other invalid strings.
The method will just catch this exception and return false because the string is an invalid format